src/pytest_plugins/filler/ported_tests.py

```diff
@@ -0,0 +1,161 @@
+showing either the static filler file paths or associated PR URLs.
+
+Usage:
+------
+# Show static filler file paths
+uv run fill --show-ported-from tests/
+
+# Show PR URLs instead
+uv run fill --show-ported-from=prs tests/
+
+The plugin will:
+1. Collect all test items with @pytest.mark.ported_from markers
+2. Extract either the file paths (first positional argument) or PR URLs (pr keyword argument)
+3. Output a deduplicated, sorted list, one per line
+4. Skip test execution (collection only)
+
+Marker Format:
+--------------
+@pytest.mark.ported_from(
+    ["path/to/static_filler1.json", "path/to/static_filler2.json"],
+    pr=[
+        "https://github.com/ethereum/execution-spec-tests/pull/1234",
+        "https://github.com/ethereum/execution-spec-tests/pull/5678",
+    ],
+)
+"""
+
+import re
+from typing import Set
+from urllib.parse import urlparse
+
+import pytest
+
+
+def convert_to_filled(file_path: str) -> str | None:
+    """Convert source link to filler to filled test path."""
+    path = urlparse(file_path).path
+    if "/src/" in path:
+        path = path.split("/src/", 1)[1]
+
+    if path.endswith((".sh", ".js")):
+        return None
+
+    # Remove "Filler" from the path components
+    path = path.replace("TestsFiller", "Tests")
+
+    # Replace file extension to .json
+    path = re.sub(r"Filler\.(yml|yaml|json)$", r".json", path)
+
+    return path
+
+
+def pytest_addoption(parser: pytest.Parser):
+    """Add command-line options to pytest."""
+    ported_from_group = parser.getgroup(
+        "ported_from", "Arguments for showing ported_from marker information"
+    )
+    ported_from_group.addoption(
+        "--show-ported-from",
+        action="store",
+        dest="show_ported_from",
+        default=None,
+        nargs="?",
+        const="paths",
+        help=(
+            "Show information from @pytest.mark.ported_from markers. "
+            "Use '--show-ported-from' or '--show-ported-from=paths' to show static filler paths. "
+            "Use '--show-ported-from=prs' to show PR URLs."
+        ),
+    )
+    ported_from_group.addoption(
+        "--links-as-filled",
+        action="store_true",
+        dest="links_as_filled",
+        default=False,
+        help=(
+            "Convert URLs or paths to filled test file paths for coverage script. "
+            "Used in combination with --show-ported-from."
+        ),
+    )
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_configure(config):
+    """Register the plugin if the CLI option is provided."""
+    if config.getoption("show_ported_from"):
+        config.pluginmanager.register(PortedFromDisplay(config), "ported-from-display")
+
+
+class PortedFromDisplay:
+    """Pytest plugin class for displaying ported_from marker information."""
+
+    def __init__(self, config) -> None:
+        """Initialize the plugin with the given pytest config."""
+        self.config = config
+        self.show_mode = config.getoption("show_ported_from")
```

Comment:
```suggestion
        self.show_mode = config.getoption("show_ported_from")
        self.links_as_filled = config.getoption("links_as_filled")
```
Just for completeness.